### PR TITLE
Feature/fetch calls

### LINKED ISSUE #6: Create backend
User Story Template

### Tasks
- [x] Create backend repo, separate from this one
- [x] Write Express server
- [x] Write GET endpoint for profiles
- [x] Write GET endpoint for single user - use for both current user and displaying profiles
- [x] Write POST endpoints for single user (use for adding appointments)
- [x] Appointments should contain unique id, ids from both users involved
- [x] Complete missing Dashboard functionality that was modularized to backend

### LINKED ISSUE #7: Create database
### Tasks
- [x] Self-teach Postgres & Knex
- [x] Move dummy data to database
- [x] Write code in Express to connect backend to database
- [x] Rewrite endpoints to replace local variables with data structures from Postgres

### PR DESCRIPTION
**Pull request by n0land0**
### Features
- Closes #6 
- Closes #7 

### Refactors 
- Update any state-setting methods to use fetch calls

### Issues and Technical Debt
- Parks data needs to be moved to the backend
- Routes should be added to Playdates to navigate to Profiles

### Other Comments, Concerns, or Questions
- n/a

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature work
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Profile/Profile.js [hunk 1]
```diff
@@ -4,54 +4,53 @@ import { Route, Switch, Link, Redirect } from 'react-router-dom'
...
-    const { appointments } = this.state
-    // const { currentUserId, selectedUserId, addPlaydateForCurrentUser } = this.props
-    // fetch (POST) new playdate for selected profile & current user
-    // fetch (GET) updated appointments for selected profile & current user
-    this.setState({ appointments: [...appointments, newPlaydate]})
-    this.props.addPlaydateForCurrentUser(newPlaydate)
+  updateProfile = () => {
+    fetchCalls.getSingleUser(+this.props.selectedUserId)
+      .then((response) => this.setState({...response}))
+      .then(() => console.log('profile', this.state))
```
Comment (PaulTimothyChambers) [inline src/Profile/Profile.js:29]: Must remove console log at some point

File: src/fetchCalls.js [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
+    return fetch('https://sniff-api.herokuapp.com/api/v1/playdates', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newPlaydates)
+    })
+    .then(response => {
+      this.checkResponse(response)
+      console.log(response)
```
Comment (PaulTimothyChambers) [inline src/fetchCalls.js:45]: Remove console log at some point

File: src/App/App.js [hunk 3]
```diff
@@ -18,30 +19,17 @@ class App extends Component {
...
-    this.setState({ currentUser: {
-      ...this.state.currentUser,
-      appointments: [...appointments, newPlaydate]
-    }})
+  updateCurrentUser = () => {
+    Promise.all([fetchCalls.getSingleUser(2), fetchCalls.getUsers()])
+    .then(([user, allUsers]) => {
+      this.setState({ currentUser: {...user}, users: allUsers })
+    })
+    .then(() => console.log('current user', this.state.currentUser))
```
Comment (PaulTimothyChambers) [inline src/App/App.js:30]: Remove console log at some point

File: src/fetchCalls.js [hunk 4]
```diff
@@ -1,22 +1,53 @@
 const fetchCalls = {
-  getUsers = () => {
-    return fetch('url')
+  getUsers() {
+    // return fetch('http://localhost:3001/api/v1/users')
```
Comment (PaulTimothyChambers) [inline src/fetchCalls.js:3]: Remove comment

**Review by PaulTimothyChambers — APPROVED** [review]

File: src/PlaydateForm/PlaydateForm.js [hunk 5]
```diff
@@ -11,7 +12,6 @@ class PlaydateForm extends Component {
   }
 
   componentDidMount = () => {
```
Comment (robbiejaeger) [inline src/PlaydateForm/PlaydateForm.js:14]: Is a `componentDidMount` necessary to set the state here? Why not set the `parks` array in state to the `parksData` that you're importing? Maybe if you're going to fetch the data and the `componentDidMount` is there for future fetch calls.

File: src/PlaydateForm/PlaydateForm.js [hunk 6]
```diff
@@ -23,13 +23,23 @@ class PlaydateForm extends Component {
...
     const usersInvolved = [ currentUserId, selectedUserId ]
-    const newPlaydate = {
-      id: Date.now(),
-      date, location, usersInvolved
-    }
-    addPlaydateForSelectedUser(newPlaydate)
+    const playdateId = Date.now()
+    const playdates = usersInvolved.map(id => {
+      return {
+        id: playdateId,
```
Comment (robbiejaeger) [inline src/PlaydateForm/PlaydateForm.js:31]: Why does each `playdate` get the same `id` with `const playdateId = Date.now()`? Do you need the `id: playdateId` to be unique in your logic?

File: src/PlaydateForm/PlaydateForm.js [hunk 7]
```diff
@@ -23,13 +23,23 @@ class PlaydateForm extends Component {
...
+        userId: id,
+        playmate: usersInvolved.find(userId => userId !== id),
+        date, location,
+      }
+    })
+
+    fetchCalls.postAppointment(playdates)
```
Comment (robbiejaeger) [inline src/PlaydateForm/PlaydateForm.js:38]: Is there any form validation before the form is submitted? You don't want to be POSTing data if the form is invalid.

File: src/fetchCalls.js [hunk 8]
```diff
@@ -1,22 +1,53 @@
...
+      this.checkResponse(response)
+      console.log(response)
+      return response.json()
+    })
+  },
 
-  checkResponse = () => {
+  checkResponse(response) {
     if (!response.ok) {
       if (response.status === 422 || response.status === 403) {
```
Comment (robbiejaeger) [inline src/fetchCalls.js:52]: Just curious, do you have this logic built into your backend API to return these status codes given certain situations? Let us know if you need help around this idea.

**Pull request merged**